src/messages/Publication.ts

```diff
@@ -14,6 +14,24 @@ export interface Publication {
   payload: any;
 }
 
+/**
+ * Handles publications
+ */
+export type PublicationHandler = (publication: Publication) => void;
+
+/**
+ * A strictly-typed event handler for publication messages.
+ */
+export declare interface PublicationEventEmitter {
```

Comment:
As we add more HostToClient messages (that need to notify the client) we would have to expand this interface using & to include WhateverEventEmitter).

Reply:
This feels like a bit much in terms of a public-facing API. 

I'd suggest that:
 - `emit` should be private from the application perspective. We don't want applications using the libraries as a general purpose eventing bus. 
- `on` and `addListener` are redundant. We should expose `on`/`off` or `add...`/`remove...` and leave it at that.
- `once` isn't really necessary and opens up a can of worms about how many eventListener helper functions do we add. I'd rather folks use other libraries and tools that can provide that functionality on top of a basic event handler
- `listenerCount` doesn't seem very useful externally.

Reply:
Yea these are fair points.  I just grabbed everything form EventEmitter without really thinking about it.

Just a point of clarification though.  JS users will still be able to get these if we expose the EventEmitter object (or extend from it).  These are just limitations for TS users.  I could always add a Facade though, if we want to restrict JS as well?

Reply:
NM, i read your comment down below.  I'll change that.